x-pack/plugins/search_playground/server/utils/fetch_indices.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IndicesIndexState } from '@elastic/elasticsearch/lib/api/types';
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+function isHidden(index: IndicesIndexState): boolean {
+  return index.settings?.index?.hidden === true || index.settings?.index?.hidden === 'true';
+}
+function isClosed(index: IndicesIndexState): boolean {
+  return (
+    index.settings?.index?.verified_before_close === true ||
+    index.settings?.index?.verified_before_close === 'true'
+  );
+}
+
+export const fetchIndices = async (
+  client: ElasticsearchClient,
+  searchQuery: string | undefined
+): Promise<{
+  indexNames: string[];
+}> => {
+  const indexPattern = searchQuery ? `*${searchQuery}*` : '*';
+  const allIndexMatches = await client.indices.get({
+    expand_wildcards: ['open'],
+    // for better performance only compute aliases and settings of indices but not mappings
+    features: ['aliases', 'settings'],
+    // only get specified index properties from ES to keep the response under 536MB
+    // node.js string length limit: https://github.com/nodejs/node/issues/33960
+    filter_path: ['*.aliases', '*.settings.index.hidden', '*.settings.index.verified_before_close'],
+    index: indexPattern,
+  });
+
+  const allIndexNames = Object.keys(allIndexMatches).filter(
```

Comment:
could we put in a page limit on number of indices? We don't want to show more than 10 in the dropdown and we allow the user to search for the index they need.

Reply:
Done